types/react-vis/index.d.ts

```diff
@@ -0,0 +1,3868 @@
+
+        opacityType?: ('linear' | 'ordinal' | 'category' | 'time' | 'time-utc' | 'log' | 'literal');
+
+        sizeBaseValue?: number;
+
+        sizeDistance?: number;
+
+        sizeDomain?: Array<number>;
+
+        sizeRange?: Array<number>;
+
+        sizeType?: ('linear' | 'ordinal' | 'category' | 'time' | 'time-utc' | 'log' | 'literal');
+
+        stack?: boolean;
+
+        style?: Object;
+
+        tickSize?: number;
+
+        width?: number;
+
+        xBaseValue?: number;
+
+        xDistance?: number;
+
+        xDomain?: Array<number | Date>;
+
+        xRange?: Array<number>;
+
+        xType?: ('linear' | 'ordinal' | 'category' | 'time' | 'time-utc' | 'log' | 'literal');
+
+        yBaseValue?: number;
+
+        yDistance?: number;
+
+        yDomain?: Array<number>;
+
+        yRange?: Array<number>;
+
+        yType?: ('linear' | 'ordinal' | 'category' | 'time' | 'time-utc' | 'log' | 'literal');
+
+
+    };
+
+    export type PropsDiscreteColorLegend = {
+
+        displayName?: string;
+
+        className?: string;
+
+        height?: number;
+
+        colors?: Array<string>;
+
+        items?: Array<string | { title: string, color: string, disabled: boolean } | Element>;
+
+        onItemClick?: ((arg0: void, arg1: Object, arg2: void, arg3: number) => any) | undefined;
+
+        onItemMouseEnter?: (datapoint: number, info: infoObject) => void;
+
+        onItemMouseLeave?: (datapoint: number, info: infoObject) => void;
+
+        orientation?: ('vertical' | 'horizontal');
+
+        width?: number;
+
+    };
+
+    export type PropsFlexibleHeightXYPlot = {
+
+        animation?: { duration: number } | boolean;
+
+        className?: string;
+
+        dontCheckIfEmpty?: boolean;
+
+        margin?: number | Object;
+
+        onClick?: (datapoint: number, info: infoObject) => void;
+
+        onDoubleClick?: (datapoint: number, info: infoObject) => void;
+
+        onMouseDown?: (datapoint: number, info: infoObject) => void;
+
+        onMouseEnter?: (datapoint: number, info: infoObject) => void;
+
+        onMouseLeave?: (datapoint: number, info: infoObject) => void;
+
+        onMouseMove?: (datapoint: number, info: infoObject) => void;
+
+        onWheel?: (e: React.WheelEvent) => void;
+
+        stackBy?: string;
+
+        style?: Object;
+
+    };
+
+    export type PropsFlexibleWidthXYPlot = {
+
+        animation?: { duration: number } | boolean;
+
+        height: number;
+
+        className?: string;
+
+        dontCheckIfEmpty?: boolean;
+
+        margin?: number | Object;
+
+        onClick?: (datapoint: number, info: infoObject) => void;
+
+        onDoubleClick?: (datapoint: number, info: infoObject) => void;
+
+        onMouseDown?: (datapoint: number, info: infoObject) => void;
+
+        onMouseEnter?: (datapoint: number, info: infoObject) => void;
+
+        onMouseLeave?: (datapoint: number, info: infoObject) => void;
+
+        onMouseMove?: (datapoint: number, info: infoObject) => void;
```

Comment:
A type alias for `(datapoint: number, info: infoObject) => void;` would also be useful.